### PR TITLE
Remove ipython pin and update to jupyter

### PR DESCRIPTION
**Pull request by damianavila**
Eventually this one should close: #2978 and #3100.

Also, it seems a pre-requisite for #3122 

Let see first what is broken...

### CODE REVIEW

**Comment by damianavila** [comment]
Update: OK, we are green, but still I need to deal with timeout (diff) and nbconverter (working locally)

**Comment by bryevdv** [comment]
looks fine at a glance. Here is the last report: https://s3.amazonaws.com/bokeh-travis/0.11.0dev1-d8b2903/report.html seems like several examples did not get images captured?

**Comment by mattpap** [comment]
> looks fine at a glance

Looking at the notebook examples, not a single seems to work. This might be just a timing issue.

**Comment by damianavila** [comment]
The notebook ones are working... the server ones are failing because we are not pointing to the correct url... I guess...

I see some examples using `output_server`, and some other not using it... what would be the standard? Because that determine which url I have to point at... (`/example_name` or `/default`)

@bryevdv @havocp 

Additionally a timeout issue could be involved as well with the server examples...

**Comment by havocp** [comment]
In a notebook you have to use output_server to output to server right? Can you point me to the code you are asking about, I probably don't understand the question.

**Comment by bryevdv** [comment]
Was I reading the report backwards? I thought the ones on the left that show notebooks with output are the current results?  It would really help me to have the columns in the report labeled.

**Comment by damianavila** [comment]
>@mattpap Was I reading the report backwards? I thought the ones on the left that *do* show notebooks with output are the current results?  It would really help me to have the columns in the report labeled.

@bryevdv  Nop, left is the latest, right is the reference, you are reading OK: https://github.com/bokeh/bokeh/blob/master/tests/travis/examples.html#L24, I can probably modify the template since I am touching these files...

>In a notebook you have to use output_server to output to server right? Can you point me to the code you are asking about, I probably don't understand the question.

@havocp, I mean the server examples, not the notebook ones...

we start the server here: https://github.com/bokeh/bokeh/blob/fix_ipython/tests/travis/test#L449-L451

and use this url: https://github.com/bokeh/bokeh/blob/fix_ipython/tests/travis/test#L346-L347

but if the example does not have the `output_server(foo)` then the `%s` is `default` (not the name of the example, in this case, `foo`) and we are pointing to the wrong url...

Additionally I think there is a timeout issue but I guess the url thing comes first...

**Comment by havocp** [comment]
Oh, it may not be clear that there are two things: the session and the app_path. Sessions are "inside" the app path. `output_server` outputs to a session ID, but it needs an app_path parameter and doesn't have it (this is on the todo list to allow specifying `app_path` when you call output_server). If you just run `bokeh serve` with no parameters, then you have an app at path `/`, and then by default we output to the session called `default`. The session name is the `?bokeh-session-id=default` thing, and the app path is just the url path like `/foo' in `http://localhost/foo`. If you omit `?bokeh-session-id` then you get a random session id (a uuid.uuid4()). `bokeh serve foo.py` defaults to app path `/foo` based on the filename.

**Comment by damianavila** [comment]
@bokeh/core 

The problem with the server examples comes from the fact that phantomjs does not support the ws protocol we use in the server (provided by tornado).

It seems not easy to fix and it is outside the scope of this PR, so my proposal is to merge this one as is and the open a new issue to track the server examples where we evaluate if we are going to invest time hacking no tornado and the bokeh server to work with the old protocol or try another alternative instead of phantomjs.

Additionally my other py35 PR depend on this one, so this should be merge this one soon.

Thoughts?

**Comment by bryevdv** [comment]
Yes agreed, let's address that in its own issue.

**Comment by damianavila** [comment]
> Yes agreed, let's address that in its own issue.

OK, I plan to merge this one tomorrow, so any review is welcome @bokeh/core

File: tests/travis/test_examples.py [hunk 1]
```diff
@@ -21,9 +22,16 @@ def test_examples(self):
...
     def test_nbexecuter(self):
         from . import nbexecuter
+
         example_dir = join(dirname(__file__), pardir, pardir, 'examples')
         example_nbconverted = join(example_dir, "glyphs", "glyph.ipynb")
-        nbexecuter.main(example_nbconverted)
+
+        kernel_name = 'python2'
+        pyver = os.environ.get('TRAVIS_PYTHON_VERSION')
+        if pyver in ['3.4', '3.5']:
```
Comment (mattpap) [inline tests/travis/test_examples.py]: This should use something more general, e.g. `pyver.startswith("3.")`, or better `split()`.

**Comment by mattpap** [comment]
A lot of changes, but looks OK. We need to clean up tests' output, because it's barely readable with all the warnings, etc.

**Comment by bryevdv** [comment]
@mattpap the warnings are now almost entirely fixed/removed. This was marked `accepted` so I am going to go ahead and merge.

**Comment by birdsarah** [comment]
> The problem with the server examples comes from the fact that phantomjs does not support the ws protocol we use in the server (provided by tornado).

We are already using firefox on travis for the selenium tests - maybe we can set something like that up for the notebook tests. If it's just a matter of running all the cells and then getting a screenshot this would be easy with selenium. If we need to test that output of the cells then that might be trickier, but nothing's impossible :D

**Comment by bryevdv** [comment]
The problem with the server example seems to be that it tries to connect to a specific session name based off the filename, but all the examples actually just publish to "default":

https://github.com/bokeh/bokeh/blob/master/tests/travis/test#L346

**Pull request merged**